### PR TITLE
feat: add nuget package creation and nuget.org package pushing

### PR DESCRIPTION
**Pull request by QuakeEye**
## Description

This pull request extends the release pipeline to create a NuGet package and upload it to https://www.nuget.org/.
The pipeline was developed on [this fork](https://github.com/QuakeEye/aplib.net-packaging-dev).

## Testability

You can check the latest release on the fork.
Additionally, [this](https://github.com/QuakeEye/aplib.net-packaging-dev/actions/runs/8775349828) is the action that succesfully ran.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch

### CODE REVIEW

File: .github/workflows/test-build-release.yml [hunk 1]
```diff
@@ -74,7 +74,8 @@ jobs:
 
   create-release:
     name: Create release
-    needs: test-build
+    needs: 
+      - test-build
```
Comment (joachimdekker) [inline .github/workflows/test-build-release.yml]: It only needs one thing, so the list can be omitted:
```suggestion
    needs: test-build
```

**Review by SSelinn — DISMISSED** [review]
Noice

File: .github/workflows/test-build-release.yml [hunk 2]
```diff
@@ -130,3 +131,76 @@ jobs:
...
+      - name: Get Latest Tag
+        id: previoustag
+        uses: WyriHaximus/github-action-get-previous-tag@v1.4.0
+        with:
+          fallback: v1.0.0
+      - name: Get version from release tag
+        id: get-version
+        run: echo "::set-output name=tag::$(echo "${{ steps.previoustag.outputs.tag }}" | cut -c 2-)"
+      - name: Create NuGet Package
+        run: dotnet pack -c Release -o ./bin --include-source --include-symbols -p:Version=${{ steps.get-version.outputs.tag }} -p:SymbolPackageFormat=snupkg
```
Comment (joachimdekker) [inline .github/workflows/test-build-release.yml]: lets also include debug symbols into the package
```suggestion
        run: dotnet pack -c Release -o ./bin --include-source --include-symbols -p:Version=${{ steps.get-version.outputs.tag }} -p:SymbolPackageFormat=snupkg -p:DebugType=embedded
```

File: .github/workflows/test-build-release.yml [hunk 3]
```diff
@@ -130,3 +131,76 @@ jobs:
...
+  upload-to-nuget-org:
+    name: Upload NuGet Package to NuGet.org
+    needs: create-nuget-package
+    runs-on: ubuntu-latest
+    permissions: write-all
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (joachimdekker) [inline .github/workflows/test-build-release.yml]: Can this be removed, as we don't use it in this stage?

**Review by joachimdekker — CHANGES_REQUESTED** [review]

File: .github/workflows/test-build-release.yml (hunk 3 above)
Comment (QuakeEye) [reply .github/workflows/test-build-release.yml]: yes

**Review by SSelinn — APPROVED** [review]
Thumbs Up

**Review by joachimdekker — APPROVED** [review]

**Pull request merged**